src/autocomplete/EmojiProvider.tsx

```diff
@@ -62,6 +67,19 @@ function score(query, space) {
     }
 }
 
+function delimiterTrimmed(string: string): string {
+    // Trim off leading and potentially trailing `:` to correctly
+    // match the emoji data as they exist in emojibase.
+    let returned = string;
+    if (string[0] === EMOJI_DELIMITER) {
+        returned = returned.substring(1);
+    }
+    if (returned[returned.length - 1] === EMOJI_DELIMITER) {
+        returned = returned.slice(0, -1);
+    }
+    return returned;
+}
```

Comment:
Per the examples included in the bug reports, some of the emoji include ':' in the middle of the shortcode. Doing that would fundamentally change the searched-for shortcode, which was part of the initial bug. I have tried to be very precise in removing only exactly the characters that have been added.

Reply:
Which emoji have `:` in the middle?

`:man:_beard:` and `:woman:_beard:` was a different bug https://github.com/vector-im/element-web/issues/19304 which are now `:man_with_beard` and `:woman_with_beard:`

Reply:
Ah, got it. I still feel that this method is better, even if not the easiest. The code shows clear intent, and does not rely on assumptions about the content of the strings the function is operating on. It maps very obviously to the code that is adding the surrounding `:`s.

My preference for code with my name on it would be leave it as is, but if it is style for this repo/project or this is the thing stopping this bugfix from getting merged then I will change it immediately.

Reply:
if we're worried about colons in the middle, why not regex instead of manually grabbing the string pieces?

Reply:
Also, I am getting issues with string.replaceAll not existing on the current typescript config. From the tsconfig file, it appears that this project targets es2016. Per https://www.typescriptlang.org/tsconfig#lib, string.replaceAll was added in es2021.

Reply:
I don't think this needs `replaceAll`, just `str.replace(/^:(.*):$/g, "$1")` (untested)